src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3006,6 +2935,90 @@ void Arguments::fix_appclasspath() {
   }
 }
 
+// arg is a JVM argument without the leading "-XX:", e.g. "+BoolOpt" or "StrOpt=str"
+bool Arguments::process_flag_for_restore(const char *arg) {
+  // -XX:Flags and -XX:VMOptionsFile do not have corresponding JVMFlags
+  if ((strncmp(arg, "Flags=", ARRAY_SIZE("Flags=") - 1) == 0) ||
+      (strncmp(arg, "VMOptionsFile=", ARRAY_SIZE("VMOptionsFile=") - 1) == 0)) {
+    return true;
+  }
+
+  const char* name;
+  size_t name_len;
+  bool has_plus_minus;
+  parse_argname(arg, &name, &name_len, &has_plus_minus);
+
+  const JVMFlag* flag;
+  {
+    char buf[BUFLEN + 1];
+    const char* stripped_name;
+    if (name[name_len] == '\0') {
+      stripped_name = name;
+    } else {
+      guarantee(name_len <= BUFLEN, "argument name too long: %s", name); // Should've been detected earlier
+      strncpy(buf, name, name_len);
+      buf[name_len] = '\0';
+      stripped_name = buf;
+    }
+    flag = JVMFlag::find_declared_flag(real_flag_name(stripped_name));
+    guarantee(flag != nullptr, "unknown JVM flag name: %s", name); // Should've been detected earlier
+  }
+  assert(has_plus_minus == flag->is_bool(), "sanity check");
+
+  if (flag->is_restore_settable()) {
+    // Restored JVM will search for the flag using the name we record here so we
+    // must ensure the real one is recorded
+    if (strcmp(flag->name(), name) == 0) {
```

Comment:
IIUC the `name` can still contain the equals sign and value? Maybe if you used the `stripped_name`...